pkg/imgpkg/bundle/bundle.go

```diff
@@ -117,7 +122,8 @@ func (o *Bundle) pull(baseOutputPath string, ui goui.UI, pullNestedBundles bool,
 		imagesLockUI = goui.NewNoopUI()
 	}
 
-	err = NewImagesLock(imagesLock, o.imgRetriever, o.Repo()).WriteToPath(filepath.Join(baseOutputPath, bundlePath), imagesLockUI)
+	err = NewImagesLock(imagesLock, o.imgRetriever, rootBundleRepo).WriteToPath(
```

Comment:
hmm using rootBundleRepo vs o.Repo() is leading me to think there is a problem here.

above we are doing 

```
subBundle := NewBundle(image.Image, o.imgRetriever)
isBundle, err := subBundle.IsBundle()
```

and im guessing that image.Image is not localized (is this true?) which then presents a problem since we would try to fetch bundle from original location. if thats true then we should be using a localized copy of the image since we want to be pulling stuff from local place and if we do initialize NewBundle with correct image, then we can continue using o.Repo() for ImagesLock().

Reply:
Oh yeah, good catch. We changed the impl to localize the image reference to the root bundle repo, prior to recursing to the next bundle.

 As for using `o.Repo()` instead of using the `rootBundleRepo`, we are going to continue to use the `rootBundleRepo` because we want to 'localize' nested bundles that are 2+ levels nested deep to the root bundle Repo. 

for more details see the above comment: https://github.com/vmware-tanzu/carvel-imgpkg/pull/109#issuecomment-805336588